Calibration/TkAlCaRecoProducers/python/ALCARECOPromptCalibProdSiStripGains_cff.py

```diff
@@ -99,7 +99,6 @@
                                             # 2 provide more detailed output
                                             Frequency = cms.untracked.int32(50),
                                             MEPathToSave = cms.untracked.string('AlCaReco/SiStripGains'),
-                                            deleteAfterCopy = cms.untracked.bool(True)
```

Comment:
on a different topic, is this supposed to be useless with the new DQM infrastructure?
Looking back in my inbox this seemed to be necessary not to explode with memory at Tier-0 during operations (but this was back in 2016).

Reply:
cf: https://github.com/cms-sw/cmssw/pull/14198#issuecomment-214433875

Reply:
I am not exactly sure what this workflow does, but in general the memory concerns at Tier0 are pretty much gone since we no longer use per-stream MEs.

If the copy in this module (which hopefully does run in a  HARVESTING step, and is not affected at all by the stream story) consumes too much memory, we need to find a different solution (like, book things in the right place, or read them from the right place, or rename things after the fact), but moving around stuff in the DQMStore is not something we can support for the future.

Actually, the comment previous to the one you linked mentioned that this is a end-of-job spike, which from legends I know was one of the reasons to move towards the DQMIO format. I wonder if we should finally stop pretending to this code that it is still run 1 and move it to current infrastructure... Sadly, I have no real clue how these workflows work and it seems nobody else has either, so that will be hard.

Reply:
@schneiml 
> the memory concerns at Tier0 are pretty much gone since we no longer use per-stream MEs.

I am not sure how relevant it is, as to the best of my knowledge Tier0 AlCaSkim and AlCaHarvest jobs are single-threaded.
I would feel much better if we could organize (for this PR and the others that will come that will change `MEtoEDMConverter`) a more meaningful check than 10 events in the PR test.  For some workflows a 10 event test will probe nothing as the computation won't even be started.
Can AlCa/DB conveners make sure this happens? @christopheralanwest @tlampen @tocheng 

>  I wonder if we should finally stop pretending to this code that it is still run 1 and move it to current infrastructure...

I would be happy to take a look to achieve this. Can you explain what you mean by "current" infrastructure?

Reply:
"current": Not `MEtoEDM`, if possible, or otherwise a good reasoning why it is required, and no reliance on `edm::Service<DQMStore>` (which implies using `DQMEDAnalyzer` or `DQMEDHarvester` base classes). It is well possible that this is not possible for some reason at the moment, but then we need to find out these reasons and decide what to do for the future.

Reply:
so, I took the liberty of doing a bit of profiling. 
I've re-run a similar step of what I had done here https://github.com/cms-sw/cmssw/pull/22515 i.e.:
```
cmsDriver.py step3 --datatier ALCARECO --conditions auto:run2_data -s ALCA:PromptCalibProdSiStripGains --eventcontent ALCARECO -n 1000 --dasquery='file dataset=/ZeroBias/Run2016C-SiStripCalMinBias-18Apr2017-v1/ALCARECO run=276097'
```
and profiled the RSS memory:

![memory](https://user-images.githubusercontent.com/5082376/68586869-f113f580-0485-11ea-9892-058419302340.png)

The spike at the end of the job is increased indeed, but I cannot comment if that is going to be a problem for Tier-0.
I also checked the output of the harvesting job, tweaking the configuration obtained via:

```
cmsDriver.py stepHarvest --data --conditions auto:run2_data --scenario pp -s ALCAHARVEST:SiStripGains --filein file:PromptCalibProdSiStripGains.root -n 1000 --fileout file:calib.root
```
in order to be forced to produce a payload via:
```python
process.alcaSiStripGainsHarvester.GoodFracForTagProd =  cms.untracked.double(0.0)   #<- fraction of good fits (default is 98%)
process.alcaSiStripGainsHarvester.NClustersForTagProd = cms.untracked.double(100000)   #<- total number of clusters (default used is 8e8)
```

and the good news is that the payload hash is the same and I don't see differences in the harvested output (full diff  [here](http://musich.web.cern.ch/musich/display/compare_SiStripGains_28247.pdf) ):
![image](https://user-images.githubusercontent.com/5082376/68587628-cb87eb80-0487-11ea-9cfc-39fb25d4d44b.png)

Reply:
Thanks @mmusich for the test!

The observation is, I'd say, exactly compatible with what is done here: During processing, the memory use is the same, but the spike while `MEtoEDM` saving is higher if the unnecessary MEs are not deleted. The absolute change seems not too big to me, not sure if Tier0 sees this the same.

That the produced payload is identical is good to know.